src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java

```diff
@@ -563,40 +570,67 @@ private boolean checkPastDateValidation(LocalDate localDate) {
             claimAmount = caseData.getRespondToAdmittedClaimOwingAmountPounds();
         }
-        BigDecimal claimFee = MonetaryConversions.penniesToPounds(caseData.getClaimFee().getCalculatedAmountInPence());
-        BigDecimal paidAmount = (caseData.getCcjPaymentPaidSomeOption() == YesOrNo.YES) ? MonetaryConversions.penniesToPounds(
-            caseData.getCcjPaymentPaidSomeAmount()) : ZERO;
-        BigDecimal subTotal = claimAmount.add(claimFee).add(caseData.getTotalInterest());
+        BigDecimal claimFee =  MonetaryConversions.penniesToPounds(caseData.getClaimFee().getCalculatedAmountInPence());
+        BigDecimal paidAmount = (caseData.getCcjPaymentDetails().getCcjPaymentPaidSomeOption() == YesOrNo.YES)
+            ? MonetaryConversions.penniesToPounds(caseData.getCcjPaymentDetails().getCcjPaymentPaidSomeAmount()) : ZERO;
+        BigDecimal fixedCost = setUpFixedCostAmount(claimAmount, caseData);
+        BigDecimal subTotal =  claimAmount.add(claimFee).add(caseData.getTotalInterest()).add(fixedCost);
         BigDecimal finalTotal = subTotal.subtract(paidAmount);
 
-        updatedCaseData.ccjJudgmentAmountClaimAmount(claimAmount);
-        updatedCaseData.ccjJudgmentAmountClaimFee(claimFee);
-        updatedCaseData.ccjJudgmentSummarySubtotalAmount(subTotal);
-        updatedCaseData.ccjJudgmentTotalStillOwed(finalTotal);
-        updatedCaseData.ccjJudgmentAmountInterestToDate(caseData.getTotalInterest());
-        updatedCaseData.ccjPaymentPaidSomeAmountInPounds(paidAmount);
+        CCJPaymentDetails ccjPaymentDetails = CCJPaymentDetails.builder()
+            .ccjJudgmentAmountClaimAmount(claimAmount)
+            .ccjJudgmentAmountClaimFee(claimFee)
+            .ccjJudgmentSummarySubtotalAmount(subTotal)
+            .ccjJudgmentTotalStillOwed(finalTotal)
+            .ccjJudgmentAmountInterestToDate(caseData.getTotalInterest())
+            .ccjPaymentPaidSomeAmountInPounds(paidAmount)
+            .ccjJudgmentFixedCostAmount(fixedCost)
+            .build();
+
+        updatedCaseData.ccjPaymentDetails(ccjPaymentDetails);
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(updatedCaseData.build().toMap(objectMapper))
+            .build();
+    }
+
+    private BigDecimal setUpFixedCostAmount(BigDecimal claimAmount, CaseData caseData) {
+        if (!YES.equals(caseData.getCcjPaymentDetails().getCcjJudgmentFixedCostOption())) {
+            return ZERO;
+        }
+        if (claimAmount.compareTo(lowerRangeClaimAmount) < 0) {
+            return lowCostAmount;
+        } else if (claimAmount.compareTo(upperRangeClaimAmount) <= 0) {
+            return midCostAmount;
+        } else {
+            return highCostAmount;
+        }
+    }
+
+    private boolean isPaidSomeAmountMoreThanClaimAmount(CaseData caseData) {
+        return caseData.getCcjPaymentDetails().getCcjPaymentPaidSomeAmount() != null
```

Comment:
same applies to this method. could be inside casedata instead?

Reply:
Updated